server/src/main/java/org/elasticsearch/index/fielddata/MultiValuedSortedBinaryDocValues.java

```diff
@@ -10,41 +10,46 @@
 package org.elasticsearch.index.fielddata;
 
 import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.DocValues;
+import org.apache.lucene.index.LeafReader;
+import org.apache.lucene.index.NumericDocValues;
 import org.apache.lucene.util.BytesRef;
 import org.elasticsearch.common.io.stream.ByteArrayStreamInput;
+import org.elasticsearch.index.mapper.MultiValuedBinaryDocValuesField;
 
 import java.io.IOException;
 
 /**
- * Wrapper around {@link BinaryDocValues} to decode the typical multivalued encoding used by
- * {@link org.elasticsearch.index.mapper.BinaryFieldMapper.CustomBinaryDocValuesField}.
+ * Wrapper around {@link BinaryDocValues} to decode the typical multivalued encoding
  */
-public class MultiValuedSortedBinaryDocValues extends SortedBinaryDocValues {
+public abstract class MultiValuedSortedBinaryDocValues extends SortedBinaryDocValues {
 
-    BinaryDocValues values;
-    int count;
+    protected final ByteArrayStreamInput in = new ByteArrayStreamInput();
```

Comment:
I think these fields can remain to be package protected? Given that subclasses are inner classes?